legend-sdlc-server/src/test/java/org/finos/legend/sdlc/server/gitlab/api/server/AbstractGitLabServerApiTest.java

```diff
@@ -0,0 +1,114 @@
+    static final String TEST_HOST_SCHEME = "https";
+    static final String TEST_HOST_HOST = "gitlab.com";
+    static final Integer TEST_HOST_PORT = null;
+    static final String TEST_HOST_URL = "https://gitlab.com";
+
+    static final BackgroundTaskProcessor backgroundTaskProcessor = new BackgroundTaskProcessor(1);
+
+    static final Logger LOGGER = LoggerFactory.getLogger(AbstractGitLabServerApiTest.class);
+
+    @BeforeClass
+    public static void suiteSetup()
+    {
+        JerseyGuiceUtils.install((s, serviceLocator) -> null);
+    }
+
+    @AfterClass
+    public static void shutDown()
+    {
+        LOGGER.info("Shutting down backgroundTaskProcessor.");
+        backgroundTaskProcessor.shutdown();
+        LOGGER.info("Shut down backgroundTaskProcessor.");
+    }
+
+    /**
+     * Authenticates to GitLab and creates a test GitLabUserContext for a project owner.
+     * Please note that for the server test, separate instances of GitLabApis are necessary for owner and member for the same test account.
+     *
+     * @return A test GitLabUserContext for a project owner.
+     * @throws LegendSDLCServerException if cannot authenticate to GitLab via OAuth.
+     */
+    protected static GitLabUserContext prepareGitLabOwnerUserContext() throws LegendSDLCServerException
+    {
+        return prepareGitLabUserContextHelper(TEST_OWNER_USERNAME, TEST_OWNER_PASSWORD);
+    }
+
+    /**
+     * Authenticates to GitLab and creates a test GitLabUserContext for a project member.
+     *
+     * @return A test GitLabUserContext for a project member.
+     * @throws LegendSDLCServerException if cannot authenticate to GitLab via OAuth.
+     */
+    protected static GitLabUserContext prepareGitLabMemberUserContext() throws LegendSDLCServerException
+    {
+        return prepareGitLabUserContextHelper(TEST_MEMBER_USERNAME, TEST_MEMBER_PASSWORD);
+    }
+
+    /**
+     * Authenticates to GitLab and creates a test GitLabUserContext.
+     *
+     * @param username the name of user for whom we create this context.
+     * @param password the password of user for whom we create this context.
+     */
+    private static GitLabUserContext prepareGitLabUserContextHelper(String username, String password) throws LegendSDLCServerException
+    {
+        return GitLabApiTestSetupUtil.prepareGitLabUserContextHelper(username, password, TEST_HOST_URL, TEST_HOST_SCHEME, TEST_HOST_HOST, TEST_HOST_PORT, LOGGER);
+    }
+
+    /**
+     * Clean up all potential test projects.
+     */
+    protected static void cleanUpTestProjects(GitLabProjectApi gitLabProjectApi)
+    {
+        List<Project> projectsToBeCleaned = gitLabProjectApi.getProjects(true, "", Lists.mutable.empty(), Lists.mutable.empty());
+        for (Project project : projectsToBeCleaned)
+        {
+            LOGGER.info("Deleting test project id: {}, name: {}", project.getProjectId(), project.getName());
+            gitLabProjectApi.deleteProject(project.getProjectId());
```

Comment:
It's a good idea to be resilient against exceptions here. Otherwise an exception when trying to delete one project can lead to failure to delete other projects.

Reply:
Fixed! Do we want to err out if failing to delete any project, though, or may we risk encountering conflicts during tests?